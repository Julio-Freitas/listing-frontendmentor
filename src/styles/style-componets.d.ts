import { Theme } from './theme';

type Theme = typeof Theme;
declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}
