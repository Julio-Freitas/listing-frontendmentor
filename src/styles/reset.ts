import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
    // this is the shared style
    html, body {
    box-sizing: border-box;
    font-size: 62.5%;
    font-family: ${({ theme }) => theme.fonts.primary.leagueSpartan};
    background-color:${({ theme }) =>
      theme.colors.secondary.lightGrayishCyanBackground};
    margin: 0 auto;

    }

    #__next {
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;
      max-width: 1140px;
      margin: 0 auto;
    }

    #__next main {
        display: flex;
        flex-direction: column;
        height: 100%;
        width: 100%;
    }
    * {
        box-sizing: border-box;
        margin: 0;
        padding: 0;
    }
    a{
        text-decoration: none;
    }

    input {
        all: unset;
    }

    button {
        cursor: pointer
    }
    img{
      display: inline-block;
      max-width:100%;
      margin:0;
    }
`;
