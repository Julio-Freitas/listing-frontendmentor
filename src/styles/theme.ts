export const Theme = {
  fonts: {
    primary: {
      leagueSpartan: 'League Spartan, sans-serif'
    },
    secondary: {
      lexendDeca: 'Lexend Deca, sans-serif'
    }
  },
  colors: {
    primary: {
      darkCyan: 'hsl(180, 29%, 50%)'
    },
    secondary: {
      lightGrayishCyanBackground: 'hsl(180, 52%, 96%)',
      lightGrayishCyanFilter: 'hsl(180, 31%, 95%)',
      darkGrayishCyanFilter: 'hsl(180, 8%, 52%)',
      veryDarkGrayinshCyan: 'hsl(180, 14%, 20%)'
    }
  }
};
