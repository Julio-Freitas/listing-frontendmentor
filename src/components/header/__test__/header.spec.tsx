import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'styled-components';
import { Theme } from '../../../styles/theme';
import { Header } from '..';

const renderWithProviderTheme = (component: React.ReactNode) => {
  return {
    ...render(<ThemeProvider theme={Theme}>{component}</ThemeProvider>)
  };
};

const mockedClickTagsHeader = jest.fn();

describe('<Card />', () => {
  it('should be render correctly', () => {
    const { container } = renderWithProviderTheme(
      <Header tagsFilter={['CSS', 'HTML']} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByRole('list-skills-header').childElementCount).toBe(2);
  });

  it('should be render correctly without tags header', () => {
    renderWithProviderTheme(
      <Header onClickTagsHeader={mockedClickTagsHeader} />
    );
    expect(screen.queryByRole('list-skills-header')).not.toBeInTheDocument();
    expect(mockedClickTagsHeader).toBeCalledTimes(0);
  });

  it('should click in some tags heaer', async () => {
    renderWithProviderTheme(
      <Header
        tagsFilter={['CSS', 'HTML']}
        onClickTagsHeader={mockedClickTagsHeader}
      />
    );
    const [cssTag] = screen.getAllByRole('listitem-skills');

    await userEvent.click(cssTag);
    expect(mockedClickTagsHeader).toHaveBeenCalledWith('CSS');
  });
});
