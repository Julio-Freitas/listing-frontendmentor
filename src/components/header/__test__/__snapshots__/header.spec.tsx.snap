// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should be render correctly 1`] = `
.c0 {
  background: hsl(180,29%,50%);
  width: 100%;
  height: 124px;
  position: relative;
  margin-bottom: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 div:first-child {
  background-image: url('/images/bg-header-desktop.svg');
  width: 100%;
  height: 100%;
  background-repeat: no-repeat;
  background-size: cover;
}

.c1 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 96%;
  background: #fff;
  padding: 1.5rem;
  bottom: -0.5rem;
  margin: 0 auto;
  gap: 2.5rem;
}

.c2 {
  font-size: 1.5rem;
  padding: 0.8rem 0.5rem;
  border-radius: 0.5rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  cursor: pointer;
  gap: 1rem;
  position: relative;
  background: hsl(180,29%,50%);
  color: hsl(180,52%,96%);
}

.c2:hover {
  border-radius: 0.5rem 0 0 0.5rem;
}

.c2:hover::after {
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
  content: '';
  background-image: url(images/icon-remove.svg);
  width: 20px;
  height: 100%;
  background-size: auto;
  background-repeat: no-repeat;
  background-color: #000;
  position: absolute;
  right: -20px;
  background-position: center;
  border-radius: 0 0.5rem 0.5rem 0;
}

<header
  class="c0"
  data-testid="header"
>
  <div />
  <ul
    class="c1"
    role="list-skills-header"
  >
    <li
      class="c2"
      role="listitem-skills"
    >
      CSS
    </li>
    <li
      class="c2"
      role="listitem-skills"
    >
      HTML
    </li>
  </ul>
</header>
`;
